### DISCUSSION TITLE
How to consume a secret from an API call

### DISCUSSION THREAD

**Discussion by marceloschmidt** [vercel/vercel#4088] [General] [answered]
We can create and delete secrets and environment variables through API, but I cannot find a way to read the secret value or the environment variable value (when it's a secret) through the API. What's the best way of doing it?

**Comment by marceloschmidt** [comment]
To add to the discussion, adding a secret + environment variable from the CLI works like a charm. But when adding through the API, after running `now env pull` I get a message saying `Error! The Secret "..." is not decryptable. (400)`.

**Comment by marceloschmidt** [comment, marked as answer]
Reading the code for the CLI, [here](https://github.com/zeit/now/blob/master/packages/now-cli/src/util/env/add-env-record.ts#L31) I finally figured it out. There's an undocumented parameter called `decryptable` that, when creating via API and setting it to true, makes it available when running `now env pull`. I still don't know if this is the correct way of using the API for secret and env variables, but looks like it works now.

**Reply by lucleray** [reply]
Yes. This is the correct way of creating secrets that are used in Development Env Variables (that can be downloaded with `now env pull`).

Secrets that are decryptable can be retrieved with an authenticated API call, so make sure to only use it when necessary!
